fossas/spectrometer
# Support for release group configuration

# Overview

This PR adds support for users who wish to associate their projects with a release group on creation.

**Important**
I made the decision to slightly deviate from the CLI v1 language here. We previously used release group name and version, however now we are using name and release. This is to better mirror what the UI shows and will be more in line with what users expect. We may have 1 or 2 users who need to update few pipelines when switching from CLI v1 to v2 due to this change. I have no idea why we used the word "version" in CLI v1.

## Acceptance criteria

- The release group flags correctly associate a project with a release group
- The release group configuration settings also work.
- If a user enters only release group release or name, the CLI fails. The API requires both.

## Testing plan

Validate that it works
1. Create a release group in a Fossa organization.
1. Run `fossa analyze` after setting both release group name and release.
1. Check that it was added to the release group.

Validate that it errors correctly
- Run `fossa analyze` and only set one of name or release.

## Risks

As mentioned above, the only risk is with differing behavior from CLI v1.

## Checklist

- [ X] I added tests for this PR's change (or confirmed tests are not viable).
- [ ] I linted and formatted (via `haskell-language-server`) any haskell files I touched in this PR.
- [ X] If this PR introduced a user-visible change, I added documentation into `docs/`. (Coming in the docs PR I opened)
- [X ] I linked this PR to any referenced GitHub issues, if they exist.
